src/Billing/AutoBilling.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+/*
+ *  package   OpenEMR
+ *  link      http://www.open-emr.org
+ *  author    Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  copyright Copyright (c )2020.. Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Billing;
+
+use Exception;
+
+class AutoBilling
+{
+    public function generateBilling($event_date, $bInfo, $pid, $userid)
+    {
+        /**
+         *  Aggregate information to fill billing table
+         */
+        $event_date = $event_date . " " . date("H:i:s"); // append current time to event date
+        $billingData = explode("-", $bInfo);  //separate up billing information into code and text
+        if (!empty($billingData[0])) {
+            $code = trim($billingData[0]);  // code that is in the calendar description
+            $text = trim($billingData[1]);  // text that is after the code in the calendar description
+        } else {
+            echo xlt("Calendar entry not properly formated");
+            die;
+        }
+        //see if there is a diagnosis in the patient chart
+        //returns an array
+        $icd10 = self::getDiagnosis($pid);
```

Comment:
I am surprised that the self:: works here since those are not static functions. Recommend changing all the `self::` calls in this class to `$this->`

Reply:
Note this then allows you to set up variables within the class such as $this->pid so then can use this within the functions rather than needing to pass the parameter.

Reply:
In the PHP documentation, it said that self can be used. However, it can only be used within the class. In the Symfony courses, I have taken. They use it this way as well.  I will change them.